### DISCUSSION TITLE
How would you represent a Family Tree in Freeplane?

### DISCUSSION THREAD

**Discussion by sifran-github** [freeplane/freeplane#1249] [General]
How would you represent a Family Tree in Freeplane?

It is not an ordinary mind map because children are children of 2 parents, not just one node. Also, if someone has children from another partner, it is a bit more complex to display.

I wonder if someone has a simple and accurate idea how to represent it in a mind map.
Maybe some combination of the new layouts can help...

**Comment by EdoFro** [comment]
@sifran-github,
I would do it this way:

![image](https://github.com/freeplane/freeplane/assets/35700575/94d0a62f-12dd-456f-8cb5-70dac3628fec)

using two user-styles:
1. descendant
2. partner (wife or husband)

and assigning child nodes layouts to each other so that:
- partner is a childnode of "descendant", but it stays close to it
   - ![image](https://github.com/freeplane/freeplane/assets/35700575/34838998-3eb9-43bd-929d-d449f3e71277)

- descendants are childnodes from a partner, and they appear to be in a next level
   - ![image](https://github.com/freeplane/freeplane/assets/35700575/06b3fb29-420a-41de-92db-6fe22513a863)

here is the example map:
[Family Tree Example.zip](https://github.com/freeplane/freeplane/files/11826279/Family.Tree.Example.zip)


Hope it helps.
edo

**Reply by sifran-github** [reply]
Thank you.
it is really good, and similar to what I had in mind.

There are several issues with it, tough, but I will just write the biggest one in my opinion:

With this design, you can represent one person (lets say, a grand-grandfather) and all his descendance. So for each of the "partners" in this design, you do not show their ancestors.
But in a full family tree, you do get to see it. For example, if you want to represent a person, his siblings, and all his ancestors from all sides of the family. mother, father, grandfathers, grandmothers. For this, I think that there needs to be a different design.

**Reply by EdoFro** [reply]
Most family trees that I know are about a Family name, where a root couple is given and all descendants and their partners and children are shown as branches of this tree. (That's why it's a "tree")
Some of them are the other way, pointing as center of the tree a person and the branches represents all his/her ancestors. That's also a tree.
I have seen combinations of both, where the roots of the tree are the ancestors of a couple and the branches are their descendants.

I can imagine you can use Freeplane to any of these cases, because they are trees.

To map all descendants and all ancestors of everyone in the map is not possible in Freeplane, because it is another data structure. It's more like a relationship map. You can have that information in a database and make different tree diagrams from the data by defining which person do you want to be the center of the map, but to use Freeplane to input all the data, I think,  it will not be the most efficient solution.

Better to look for a real Family /relationship software.

**Reply by FabyoAW** [reply]
> @sifran-github, I would do it this way:
> 
> ![image](https://private-user-images.githubusercontent.com/35700575/247762450-94d0a62f-12dd-456f-8cb5-70dac3628fec.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg2NjQ1MDIsIm5iZiI6MTcyODY2NDIwMiwicGF0aCI6Ii8zNTcwMDU3NS8yNDc3NjI0NTAtOTRkMGE2MmYtMTJkZC00NTZmLThjYjUtNzBkYWMzNjI4ZmVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDExVDE2MzAwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM1OTFmNmFhYzhjNDRiYTkyMDdlZTBkNjNiMzJhYWJkOTM4Y2JhOGJjZGRjN2EzZGU3N2UxZmI5MjM2YzJkOTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.-fSRJ5VhmD82bom7nqwjGMycjZ8UelPz2-mQX5aRMyA)
> 
> using two user-styles:
> 
> 1. descendant
> 2. partner (wife or husband)
> 
> and assigning child nodes layouts to each other so that:
> 
> * partner is a childnode of "descendant", but it stays close to it
>   
>   * ![image](https://private-user-images.githubusercontent.com/35700575/247764143-34838998-3eb9-43bd-929d-d449f3e71277.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg2NjQ1MDIsIm5iZiI6MTcyODY2NDIwMiwicGF0aCI6Ii8zNTcwMDU3NS8yNDc3NjQxNDMtMzQ4Mzg5OTgtM2ViOS00M2JkLTkyOWQtZDQ0OWYzZTcxMjc3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDExVDE2MzAwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ4YjhiMzYzMDViNDc4YjcyYTMwNjA1M2U5NzgyMTlkMjYzMjljZmZmMTIzZmQzMTEzYWRjNzViMGE0NzIzMzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.bmYo0Lb1ZjhQOHUJw0F5jlAfTaAUCl-5aGlFXks114c)
> * descendants are childnodes from a partner, and they appear to be in a next level
>   
>   * ![image](https://private-user-images.githubusercontent.com/35700575/247764385-06b3fb29-420a-41de-92db-6fe22513a863.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg2NjQ1MDIsIm5iZiI6MTcyODY2NDIwMiwicGF0aCI6Ii8zNTcwMDU3NS8yNDc3NjQzODUtMDZiM2ZiMjktNDIwYS00MWRlLTkyZGItNmZlMjI1MTNhODYzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDExVDE2MzAwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBkMjIwZmQ5MGE3N2RjMmZlNmJmZmFmOGY2M2MwNGMzYWUxYmViY2U4YjExYzhmYmE4MjAyYTViMGFmODE5MzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.7lM2QU-tdjdtIiHfaMp8h2CL5rrlOLO4cUl2BaM6VEw)
> 
> here is the example map: [Family Tree Example.zip](https://github.com/freeplane/freeplane/files/11826279/Family.Tree.Example.zip)
> 
> Hope it helps. edo

It seems great, but every descendant has two parents (known or not), which may or may not be the couple linked to it. Example, the child can be descendant only of the mother, not of her husband, and the graphic should show this precisely. So, each descendant should have the possibility to be linked to two ancestor...

**Comment by sifran-github** [comment]
I'm actually really questioning whether it can be represented with a mind map.

Mindmaps are built so there is one root node, and like I said in the pervious message to @EdoFro , If you want to represent both ancestors and descendance (lets say, the people who are close to you), then no side of the family tree will be a single node. in the bottom there are many children, and in the top, there are many grand-grandparents.

In @EdoFro's example, there is only ONE grand-grandparent and all his descendance, which is possible to do, but its not a complete tree.

Here is an example of a family tree i built using https://www.familyecho.com/
![image](https://github.com/freeplane/freeplane/assets/5730355/ae55e7d5-da5c-4f34-bc92-b25baecfdf0d)
I didn't even add the complexity of different children from different partners, and I'm having really hard time trying to come up with some representation of this data in a mind map.

maybe I miss some basic idea? or maybe its just impossible?

---

EDIT: To make it more visual:

with @EdoFro's design, if you choose a specific ancestor, the following people can be represented in the mind map:

![image](https://github.com/freeplane/freeplane/assets/5730355/42f0cc2f-4e66-44b5-9e8e-c01ab928d387)

And its also possible to build a reverse mind map to represent myself and all my ancestors:
![image](https://github.com/freeplane/freeplane/assets/5730355/bd19e043-1b54-418f-9e87-1a108bd5a486)

but both together?

**Reply by EdoFro** [reply]
that's the way I would do it:
![image](https://github.com/freeplane/freeplane/assets/35700575/75ecd8d2-f554-46b2-a931-00d768dbb2fc)

If it's horizontal or vertical, that's just a matter of format.

**Reply by EdoFro** [reply]
and if you select the root node and apply this layout: ![image](https://github.com/freeplane/freeplane/assets/35700575/ad7c706f-b806-41d5-bae1-f8585b0a4ff0)

then it looks like this:
![image](https://github.com/freeplane/freeplane/assets/35700575/6f7453f9-4e74-481a-b260-3f2a8a41befb)

**Reply by sifran-github** [reply]
it is very interesting. the best version that I have seen so far.

I see the "trick" that allowed it:
1. A tree from you to your ancestors.
2. Another tree from your "family" to descendants (+ having a wide base, which is a key part of the "trick").

The most difficult part in this map is to represent sideways (uncles), and the way you chose to deal with is to have the base of the second tree wide enough to cover the immediate ones.

I think that it does get harder if you want to add more uncles to the map (sideways, people who are not directly up or down from you):
1. first cousins (one level up -> sideways -> one level down): you cover it by the wide base base of the second tree.
2. second cousins (two levels up -> sideways -> two levels down): how would you add them? I guess you could add more nodes to the base of the second tree? next to "my fathers siblings" you could add "my second cousin" and have the entire tree down from there.
3. Third cousins.... even harder. but i guess you could just add more and more nodes at the same level.

The only thing that will be missing is: if you expand sideways to cousins this way, its hard to represent the connection from them to you (because the connection is through common ancestors and the second tree is all about descendants).

but even with this limitation, its the best version that I have seen so far.

**Reply by sifran-github** [reply]
and also, how would you add another sideways (siblings) item: your grandfather's brother?

**Reply by EdoFro** [reply]
Sorry, I don't know. Maybe using connectors and their "simulate Edge" formatting.

BR

edo

**Reply by Ahmed-Al-Imari** [reply]
> and if you select the root node and apply this layout: ![image](https://private-user-images.githubusercontent.com/35700575/247786543-ad7c706f-b806-41d5-bae1-f8585b0a4ff0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA2MDk2ODksIm5iZiI6MTczMDYwOTM4OSwicGF0aCI6Ii8zNTcwMDU3NS8yNDc3ODY1NDMtYWQ3YzcwNmYtYjgwNi00MWQ1LWJhZTEtZjg1ODViMGE0ZmYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTAzVDA0NDk0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWExYWI5MzEwNWQyNDZjYTNmMTRmMGFmZDgxY2UxNTMzMWFjOTQyOGIxZTYyMDkwMmIzZDVmNTQwMDc1ZmFkMDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.tWcwXCRF96WyOKirR9jhI2aJnVsgPsrKfqUP4_02-14)
> 
> then it looks like this: ![image](https://private-user-images.githubusercontent.com/35700575/247786781-6f7453f9-4e74-481a-b260-3f2a8a41befb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA2MDk2ODksIm5iZiI6MTczMDYwOTM4OSwicGF0aCI6Ii8zNTcwMDU3NS8yNDc3ODY3ODEtNmY3NDUzZjktNGU3NC00ODFhLWIyNjAtM2YyYThhNDFiZWZiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTAzVDA0NDk0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQwNGFkYTkzY2FiZTk4NWZlOGEwZGRhYmFmZjM2MmZiNWVlMTcyNmU4MGU5ZGQ4ZWFhNGZjZDIxZDUxZDA3NzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.lGJ6pAZWJfqLAbGPrqNgqgpkhQV6Y8kD-AD3KRx_22c)

Hello,
Do you have a .mm zip file for this horizontal arrangement, please.
Thanks.
